### PR TITLE
feat(datasource/ansible-galaxy): use pep440 versioning by default

### PR DESCRIPTION
**Pull request by nejch**
## Changes

~~Parses a `galaxy.yml` dependency into a valid version string even when defined with some of the python-semanticversion identifiers.~~
~~This improves support a bit, but doesn't actually include all possibilities. I'm not sure this is really the right place to extract `currentVersion` like that, but since I see already now it's only extracted via regex so it's a bit limited, I thought it'd be a start. Happy to change it though.~~

Uses pep440 versioning by default for the galaxy datasource to support version specifiers.

/cc @dlouzan

## Context

Galaxy dependencies can be defined with version specifiers:
https://docs.ansible.com/ansible/latest/dev_guide/collections_galaxy_meta.html#structure

Using galaxy's version parser based on python-semanticrelease:
https://github.com/ansible/ansible/blob/8b2e6285650ec42ec4a19075a8567047e8304ba2/lib/ansible/galaxy/dependency_resolution/versioning.py#L26-L35

## Documentation (please check one with an [x])

- [ ] I have updated the documentation, or
- [x] No documentation update is required

## How I've tested my work (please tick one)

I have verified these changes via:

- [ ] Code inspection only, or
- [x] Newly added/modified unit tests, or
- [ ] No unit tests but ran on a real repository, or
- [ ] Both unit tests + ran on a real repository

### CODE REVIEW

File: lib/modules/manager/ansible-galaxy/util.ts [hunk 1]
```diff
@@ -5,7 +5,7 @@ export const blockLineRegEx = /^\s*((\w+):\s*(.*))$/;
 export const galaxyDepRegex = /[\w-]+\.[\w-]+/;
 export const dependencyRegex = /^dependencies:/;
 export const galaxyRegEx = regEx(
-  /^\s+(?<packageName>[\w.]+):\s*["'](?<version>.+)["']\s*/
+  /^\s+(?<packageName>[\w.]+):\s*["'](([=<>])=)?(?<version>.+)["']\s*/
```
Comment (viceice) [inline lib/modules/manager/ansible-galaxy/util.ts:8]: i think those specifiers need to be part of the version, so renovate can detect them as version range.

we probably need another versioning here. looks like pep440 could match?

**Review by viceice — CHANGES_REQUESTED** [review]

File: lib/modules/manager/ansible-galaxy/util.ts (hunk 1 above)
Comment (viceice) [reply lib/modules/manager/ansible-galaxy/util.ts:8]: or it's more like the npm versioning (partly) 🤔

Comment (nejch) [reply lib/modules/manager/ansible-galaxy/util.ts:8]: @viceice would it be too aggressive to just place a `defaultVersioning` for ansible-galaxy with pep440 versioning here?

https://github.com/renovatebot/renovate/blob/b6670159c4c6c3e6e1b07829604b2c3301f54162/lib/modules/manager/ansible-galaxy/index.ts#L7-L9

Comment (viceice) [reply lib/modules/manager/ansible-galaxy/util.ts:8]: It should better be places on the datasource.

https://github.com/renovatebot/renovate/blob/638302e519a3c76c01f185b22a4d05289f054b9f/lib/modules/datasource/datasource.ts#L25

you need to override that like his

https://github.com/renovatebot/renovate/blob/638302e519a3c76c01f185b22a4d05289f054b9f/lib/modules/datasource/docker/index.ts#L379

Comment (nejch) [reply lib/modules/manager/ansible-galaxy/util.ts]: Thanks @viceice I changed that. Sorry for the force-push, I reverted the wrong commit :facepalm: but also it's a fresh start for the PR as none of the original changes were there.

Should there be any tests added for this? From what I can see, no pep440-specific tests are done on the datasources that use them, more on the versioning side.

**Review by viceice — APPROVED** [review]

**Comment by viceice** [comment]
@rarkins should this be a feature?

**Pull request merged**